tests/csm/rest/test_iam_users.py

```diff
@@ -2127,6 +2126,190 @@ def test_39217(self):
+        self.created_iam_users.add(usr1["user_id"])
+
+        self.log.info("Step-2: Creating IAM user 2")
+        payload = self.csm_obj.iam_user_payload_rgw("valid")
+        resp = self.csm_obj.create_iam_user_rgw(payload)
+        self.log.info("Verify Response : %s", resp)
+        assert resp.status_code == HTTPStatus.CREATED, "IAM user creation failed"
+        usr2 = resp.json()
+        self.created_iam_users.add(usr2["user_id"])
+
+        self.log.info("Step-3: Edit IAM user with access key of user 1")
+        payload = {"access_key": usr1["keys"][0]["access_key"],
+                    "secret_key":usr1["keys"][0]["secret_key"]}
+        resp = self.csm_obj.modify_iam_user_rgw(usr2["user_id"], payload)
+        assert resp.status_code == HTTPStatus.CONFLICT, "PATCH status code check failed"
+        resp = resp.json()
+        if CSM_REST_CFG["msg_check"] == "enable":
+            err_code = self.csm_conf["test_38969"]["error_code"]
+            assert int(resp["error_code"]) == err_code, "Error code check failed"
+            msg_id = self.csm_conf["test_38969"]["message_id"]
+            assert resp["message_id"] == msg_id, "Message id check failed"
+            msg = self.rest_resp_conf[err_code][msg_id][0]
+            assert resp["message"] == msg, "Message check failed"
+
+        self.log.info("##### Test completed -  %s #####", test_case_name)
+
+
+    @pytest.mark.csmrest
+    @pytest.mark.lc
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-38918')
+    def test_38918(self):
+        """
+        Verify IO operations fails with old key after secret key is modified
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("Step-1: Creating IAM user 1")
+        payload = self.csm_obj.iam_user_payload_rgw("valid")
+        resp = self.csm_obj.create_iam_user_rgw(payload)
+        self.log.info("Verify Response : %s", resp)
+        assert resp.status_code == HTTPStatus.CREATED, "IAM user creation failed"
+        usr = resp.json()
+        self.created_iam_users.add(usr["tenant"] + "$" + usr["user_id"])
+        akey = usr["keys"][0]["access_key"]
+        skey = usr["keys"][0]["secret_key"]
+        bucket = "testbucket"
+        test_file = "test-object.txt"
+
+        s3_obj = S3TestLib(access_key = akey, secret_key=skey)
+        resp = s3_obj.create_bucket(bucket)
+        size = SystemRandom().randrange(10, 100)
+        file_path_upload = os.path.join(TEST_DATA_FOLDER, test_file)
+        if os.path.exists(file_path_upload):
+            os.remove(file_path_upload)
+        system_utils.create_file(file_path_upload, size)
+        resp = s3_obj.put_object(bucket_name=bucket, object_name=test_file,
+                    file_path=file_path_upload)
+
+        payload = {"access_key": usr["keys"][0]["access_key"],
+                    "secret_key":"sdss"}
```

Comment:
Can this secret key be fetched from config instead of hardcoding it?

Reply:
Replaced it with random key generated.